v0/destinations/freshsales/utils.js

```diff
@@ -39,44 +81,284 @@ const getUserAccountDetails = async (userEmail, Config) => {
+const getContactsDetails = async (userEmail, Config) => {
   const requestOptions = {
     headers: {
       Authorization: `Token token=${Config.apiKey}`,
       "Content-Type": "application/json"
     }
   };
-  const endPoint = `https://${Config.domain}${CONFIG_CATEGORIES.GROUP.endpoint}`;
-  let accountResponse = await httpPOST(endPoint, payloadBody, requestOptions);
-  accountResponse = processAxiosResponse(accountResponse);
-  if (accountResponse.status !== 200 && accountResponse.status !== 201) {
-    const errMessage = accountResponse.response?.errors?.message || "";
-    const errorStatus = accountResponse.response?.errors?.code || 500;
+  const userPayload = {
+    unique_identifier: {
+      emails: userEmail
+    },
+    contact: {
+      emails: userEmail
+    }
+  };
+  const endPoint = `https://${Config.domain}${CONFIG_CATEGORIES.IDENTIFY.baseUrl}`;
+  let userResponse = await httpPOST(endPoint, userPayload, requestOptions);
+  userResponse = processAxiosResponse(userResponse);
+  if (userResponse.status !== 200 && userResponse.status !== 201) {
+    const errMessage = userResponse.response.errors?.message || "";
+    const errorStatus = userResponse.response.errors?.code || "500";
     throw new CustomError(
-      `failed to create/update group ${errMessage}`,
+      `failed to fetch Contact details ${errMessage}`,
       errorStatus
     );
   }
-  return accountResponse;
+  return userResponse;
+};
+
+/*
+ * This function is used building the response with the user Details
+ * @param {*} email - email for Contact details
+ * @param {*} Config - headers, apiKey...
+ * @param {*} payload - created after transformation
+ * @param {*} salesActivityTypeId - for creating the sales activity
+ * returns
+ */
+
+const responseBuilderWithContactDetails = async (
+  email,
+  Config,
+  payload,
+  salesActivityTypeId
+) => {
+  const userDetails = await getContactsDetails(email, Config);
+  const userId = userDetails.response.contact.id;
+  const responseBody = {
+    ...payload,
+    targetable_id: userId,
+    sales_activity_type_id: salesActivityTypeId
+  };
+  return responseBody;
+};
+
+/*
+ * This function is used for creating sales Activity of Contact.
+ * @param {*} payload - created after transformation
+ * @param {*} message - input message
+ * @param {*} Config - headers, apiKey...
+ * ref: https://developers.freshworks.com/crm/api/#list_all_sales_activities
+ */
+const UpdateContactWithSalesActivity = async (payload, message, Config) => {
+  const requestOptions = {
+    headers: {
+      Authorization: `Token token=${Config.apiKey}`,
+      "Content-Type": "application/json"
+    }
+  };
+  if (!payload.sales_activity_name && !payload.sales_activity_type_id) {
+    throw new CustomError(
+      `Either of sales activity name or sales activity type id is required. Aborting!`,
+      400
+    );
+  }
+
+  const email = get(message, "email");
+
+  if (!payload.targetable_id && !email) {
```

Comment:
targetable_id* 	number 	ID of the contact/deal/account related to the sales activity created

looks like `targetable_id` is an internal id of freshsales shouldn't we map this to our externalId?

In rudderPayload example: 
.....
externalId: {
    type: "freshworks_contact"
    id: 123
}

Reply:
Done